crates/pallets/starknet/src/lib.rs

```diff
@@ -1178,13 +1178,13 @@ impl<T: Config> Pallet<T> {
                 state_tree.set(storage_key, storage_value);
             });
 
-            // Update the state trie for this contract in runtime storage.
-            ContractTries::<T>::set(contract_address, Some(state_tree.clone()));
-
             // We then compute the state root
             // And update the storage trie
             let state_root = state_tree.commit();
 
+            // Update the state trie for this contract in runtime storage.
+            ContractTries::<T>::set(contract_address, Some(state_tree.clone()));
+
```

Comment:
Why moving it after?

Reply:
Otherwise the state trie stored in runtime was not committed

Reply:
So it's a bug fix?

Reply:
Yes